plenum/server/consensus/ordering_service.py

```diff
@@ -2166,6 +2171,25 @@ def _caught_up_till_3pc(self, last_caught_up_3PC):
         self.last_ordered_3pc = last_caught_up_3PC
         self._remove_till_caught_up_3pc(last_caught_up_3PC)
 
+        # Get all pre-prepares and prepares since the latest stable checkpoint
+        ledger = self.db_manager.get_ledger(AUDIT_LEDGER_ID)
+
+        to = ledger.get_last_txn()['txn']['data']['ppSeqNo']
+        frm = to - to % self._config.CHK_FREQ
+
+        batch_ids = [
+            BatchID(
+                view_no=self.view_no,  # TODO: Nemanja test passes if I put `txn['txn']['data']['viewNo']` instead of `self.view_no`. Why?
```

Comment:
> TODO: Nemanja test passes if I put `txn['txn']['data']['viewNo']` instead of `self.view_no`. Why?

Yeah, I think this is because there is the following order of actions after catchup currently (see `allLedgersCaughtUp`):
- call `replica.on_catch_up_finished` (which in turn calls this method)
- call `select_primaries_on_catchup_complete` which restores viewNo.
So, `self.view_no` is always 0 at this line.

I think setting it to `get_payload_data(txn)[AUDIT_TXN_VIEW_NO]` should be fine and correct

Reply:
ok, tnx!